core/src/cache.rs

```diff
@@ -323,6 +324,10 @@ pub enum SourceState {
 
 impl Cache {
     pub fn new(error_tolerance: ErrorTolerance) -> Self {
+        let import_paths = std::env::var("NICKEL_PATH")
+            .map(|paths| paths.split(':').map(PathBuf::from).collect())
+            .unwrap_or_default();
```

Comment:
I feel like this should rather be an argument to `Cache::new` and then the actual logic from where to get the path should be up to the particular use case.

Reply:
Putting this in `Cache::new` required threading them through a lot of things, so I tried out making the import paths mutable after construction. This changes the behavior a little, but I think it might be desirable actually: when running `nix export file.nix`, the path "file.nix" is now only searched for in the current directory, and `NICKEL_PATH` is only used for finding things that "file.nix" imports.